src/entities/Vault.ts

```diff
@@ -729,4 +712,48 @@ export class Vault extends Entity {
       )
     )
   }
+
+  /**
+   * Get the maximum reserve for this vault.
+   * @internal
+   */
+  _maxReserve() {
+    return this._query(['maxReserve'], () =>
+      combineLatest([this._root._protocolAddresses(this.chainId), this._investmentCurrency()]).pipe(
+        switchMap(([{ syncManager }, asset]) =>
+          defer(async () => {
+            const maxReserve = await this._root.getClient(this.chainId).readContract({
+              address: syncManager,
+              abi: ABI.SyncRequests,
+              functionName: 'maxReserve',
+              args: [this.pool.id.raw, this.shareClass.id.raw, this._asset, 0n],
+            })
+            return new Balance(maxReserve, asset.decimals)
+          })
+        )
+      )
+    )
+  }
+
+  /**
+   * Get the available balance for this vault from the BalanceSheet.
+   * @internal
+   */
+  _availableBalance() {
+    return this._query(['availableBalance'], () =>
+      combineLatest([this._root._protocolAddresses(this.chainId), this._investmentCurrency()]).pipe(
+        switchMap(([{ balanceSheet }, asset]) =>
+          defer(async () => {
+            const availableBalance = await this._root.getClient(this.chainId).readContract({
+              address: balanceSheet,
+              abi: ABI.BalanceSheet,
+              functionName: 'availableBalanceOf',
+              args: [this.pool.id.raw, this.shareClass.id.raw, this._asset, 0n],
+            })
+            return new Balance(availableBalance, asset.decimals)
+          })
```

Comment:
needs `repeatOnEvents` with `NoteDeposit` and `Withdraw` events on the BalanceSheet